### PR TITLE
Adds type casting as per US40254

### PR DESCRIPTION
**Pull request by asauterChicago**
This is for: https://rally1.rallydev.com/#/53007768615d/detail/userstory/170323526260

### CODE REVIEW

**Comment by asauterChicago** [comment]
We don't have dev approved tags here, so I just need to devs to give a thumbs up, then in the official we'll do another approval with the updated module.

File: src/Entity/PropertyMapper.php [hunk 1]
```diff
@@ -337,6 +351,123 @@ protected function checkAnding(array &$context, array $mergedArray = []) {
...
+  /**
+   * Checks and sets any typecasting. This will override any final values.
+   *
+   * @param string $source_prop
+   * @param string $dest_prop
+   */
+  protected function checkTypeCasting(&$source_prop, $dest_prop){
+    preg_match("/!!\b([a-z]*){2,}\b/", $source_prop, $matches);
+    if (count($matches) > 0) {
+      $type = str_replace('!!', '', $matches[0]);
```
Comment (scottrigby) [inline src/Entity/PropertyMapper.php]: @asauterChicago would you also add example type casting on this function, similar to what's in the comment block of `checkOring()`?

**Review by scottrigby — APPROVED** [review]
LGTM 🏆  I've wanted to add some type-casting functionality for a while. Honestly would love to also now deprecate `hook_aws_sqs_entity_normalized_value_alter()`, because I only ever used it for type casting. But as I understand others are now using that for other logic, so… oh well ¯\_(ツ)_/¯ Still nice addition

**Comment by asauterChicago** [comment]
@scottrigby I don't think you need to deprecate the hook. The hook is still useful, for example, we use it for all kinds of defaults like "is_int? then cast as integer". However, some cases we actually want the int to be casted as a string. And rather edit the hook every time for every possible destination and source that needs special handling, this provides an override that forces the value to be something else in those special cases. 

I think the hook is still useful, in most cases the hook should probably be preferred over junking up every YAML file with "!!int or !!string".

**Comment by scottrigby** [comment]
@asauterChicago So it sounds like having both options is good because there they can represent whitelist/blacklist patterns, with flexibility depending on the required typing from whatever service is ingesting the queued messages. That's a good enough use case for me to leave the hook 👍

**Pull request merged**